src/ui/server/helpers/get-valid-fields/index.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * getValidFields
+ * Strip invalid fields from submitted form data
+ * @param {Object} Form data
+ * @param {Object} Errors list
+ * @returns {Object} Form data with invalid fields removed
+ */
+const getValidFields = (formData: object, errorList: object) => {
+  if (formData && errorList) {
+    const fieldsWithErrors = Object.keys(errorList);
+
+    const validFields = {};
+
+    Object.keys(formData).forEach((fieldName) => {
+      if (!fieldsWithErrors.includes(fieldName)) {
+        validFields[fieldName] = formData[fieldName];
+      }
+    });
+
+    return validFields;
+  }
+
+  return {};
```

Comment:
if there is no error list, shouldn't all the fieldNames be returned?

Reply:
Excellent spot! Will fix

Reply:
Added this in